x-pack/legacy/plugins/integrations_manager/server/config.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import JoiNamespace from 'joi';
+
+export const config = {
+  schema: schema.object({
+    enabled: schema.boolean(),
+    registryUrl: schema.string(),
+  }),
+};
+
+export type IntegrationsManagerConfigSchema = TypeOf<typeof config.schema>;
+
+// This is needed for the legacy plugin / NP shim setup in ../index
+// It has been moved here to keep the two config schemas as close together as possible.
+// Once we've moved to NP, the Joi version will disappear, so we're not trying to generate
+// these two config schemas from a common definition or from each other.
+
+export const getConfigSchema = (Joi: typeof JoiNamespace) => {
+  const IntegrationsManagerConfigSchema = Joi.object({
+    enabled: Joi.boolean().default(true),
+    registryUrl: Joi.string()
+      .uri()
+      .default(),
+  }).default();
+
+  return IntegrationsManagerConfigSchema;
+};
+
+const DEFAULT_CONFIG = {
+  enabled: true,
+  registryUrl: 'http://integrations-registry.app.elstc.co',
+};
+
+// As of 2019, this is a Singleton because of the way JavaScript modules are specified.
```

Comment:
Yes.

Reply:
> I'm not clear about this message

Why is that of interest or concern? What's this risk/impact/etc

Reply:
It is meant to answer the question why this works in the first place, and why this is safe enough for now, to make it easier for new people, or me in a few months, to understand this piece of code.

We're bypassing the proper flow of data (down from the handlers through function parameters) by putting something into global scope. For me, that warrants at least a comment, it's the thing I typically stumble upon while trying to understand large code bases. It's also something that wouldn't necessarily work in other languages.

Given that comments are stripped out during build, does it hurt to keep it?

Reply:
> to make it easier for new people, or me in a few months, to understand this piece of code.

I'm asking questions because they didn't clear things up for me. The comments themselves confuse(d) me; not the code. I can tell they are trying to tell me something but I'm not sure what. Especially why this one places gets a "this is how modules work" when we use them throughout the project. Why is this one getting a comment?

> We're bypassing the proper flow of data (down from the handlers through function parameters) by putting something into global scope. For me, that warrants at least a comment, it's the thing I typically stumble upon while trying to understand large code bases. It's also something that wouldn't necessarily work in other languages.

This is helpful, and could also be included, thanks.

I'm curious what you mean by global scope. Other scripts/modules can't access anything here directly, right? They must `import` this file.

Is the concerning/interesting thing here that the object isn't the live stream/observable but rather an object that _should_ be up-to-date but may not be?

> Given that comments are stripped out during build, does it hurt to keep it?

I'm not talking about file size. You have my permission to ignore me if I ever say, "These comments are too many bytes to ship".